### PR TITLE
Normalization fix Snowflake Prefix Tables starting with number

### LINKED ISSUE #9259: Snowflake Normalization doesn't work with table prefix starting with numbers
## Environment
- **Airbyte version**:  0.35.2-alha
- **OS Version / Instance**: GCP
- **Deployment**: Docker
- **Source Connector and version**: All Sources
- **Destination Connector and version**: Snowflake
- **Severity**: Very Low / Low / Medium / **High** / Critical
- **Step where error happened**: Sync Job in Normalization Step

## Current Behavior
With incremental normalization running Snowflake destination with table prefix starting with numbers break the code.
This is caused because:
https://github.com/airbytehq/airbyte/blob/dfebddc07656f3d2012179a1b6cd40be43599fbe/airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py#L1000-L1012

When creating a connection with prefix, eg, `1_stg_table` the code will generate a post-hook for Snowflake as:
```
drop view schema.1_stg_table'
```
which is invalid.

## Expected Behavior
Correct the normalization to work with number table name prefix.

## Logs
*If applicable, please upload the logs from the failing operation. 
For sync jobs, you can download the full logs from the UI by going to the sync attempt page and 
clicking the download logs button at the top right of the logs display window.*

<details>
<summary>LOG</summary> 

```

replace this with
your long log
output here

```

</details>

## Steps to Reproduce
1.
2.
3.

## Are you willing to submit a PR?

Remove this with your answer.

### PR DESCRIPTION
**Pull request by marcosmarxm**
## What
Closes https://github.com/airbytehq/airbyte/issues/9259

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>

#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py [hunk 1]
```diff
@@ -1034,6 +1034,8 @@ def add_to_outputs(
             if suffix == "scd":
                 stg_schema = self.get_schema(True)
                 stg_table = self.tables_registry.get_file_name(schema, self.json_path, self.stream_name, "stg", truncate_name)
+                if stg_table[0].isdigit():
+                    stg_table = f'"{stg_table}"'
```
Comment (ChristopheDuong) [inline airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py:1038]: Why is the fix only on "stg" tables not on standard table names too? 

If I understand, you are forcing quoting into the SQL model file name... I'm not sure why do you take this approach though?
Does this apply regardless of the destination type?

Shouldn't this be configured through https://docs.getdbt.com/reference/project-configs/quoting maybe?

the get_file_name method is making a final call to `normalize_table_name`, shouldn't this digit check happen there instead?

**Review by ChristopheDuong — CHANGES_REQUESTED** [review]

File: airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/postgres/test_simple_streams/dbt_project.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ profile: 'normalize'
 # These configurations specify where dbt should look for different types of files.
 # The `source-paths` config, for example, states that source models can be found
 # in the "models/" directory. You probably won't need to change these!
-source-paths: ["modified_models"]
+source-paths: ["models"]
```
Comment (ChristopheDuong) [inline airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/postgres/test_simple_streams/dbt_project.yml:18]: do you know why these are changing?

Comment (marcosmarxm) [reply airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/postgres/test_simple_streams/dbt_project.yml:18]: No, I didn't change it directly!

File: airbyte-integrations/bases/base-normalization/normalization/transform_catalog/destination_name_transformer.py [hunk 3]
```diff
@@ -150,6 +150,11 @@ def __normalize_non_column_identifier_name(
         if truncate:
             result = self.truncate_identifier_name(input_name=result, conflict=conflict, conflict_level=conflict_level)
         result = self.__normalize_identifier_case(result, is_quoted=False)
+        if result[0].isdigit():
+            if self.destination_type == DestinationType.MSSQL:
+                result = "_" + result
+            elif self.destination_type == DestinationType.ORACLE:
+                result = "ab_" + result
```
Comment (marcosmarxm) [inline airbyte-integrations/bases/base-normalization/normalization/transform_catalog/destination_name_transformer.py:157]: @ChristopheDuong I need to made this to work with Oracle and MSSQL. Oracle doesn't allow tables to start with `_` so I added the prefix `ab`. What do you think?

**Review by ChristopheDuong — APPROVED** [review]

**Pull request merged**